src/api/chat.js

```diff
@@ -48,29 +63,22 @@ const getChatLines = async ({ KuskiIndex, start, end, offset }) => {
       },
     ],
     where,
-    offset,
+    offset: offset ? searchOffset(offset) : 0,
+    limit: CHAT_API_LIMIT,
   };
 
-  const lines = await Chat.findAll(opts);
+  if (limit < CHAT_API_LIMIT) opts.limit = limit;
+
+  if (order === 'DESC') opts.order = [['ChatIndex', 'DESC']];
+
+  const lines = await Chat.findAndCountAll(opts);
```

Comment:
If you do just findAll here the db query is litterally a 100 times faster. I know you need the count for the pagination, but maybe worth considering doing that a different way?

Reply:
I'm thinking I could count only on the first request. Dunno how to do pagination without either doing at least one count or doing infinite scroll instead.